University-advisor-web/Controllers/SettingsController.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using University_advisor_web.Models;
+
+namespace University_advisor_web.Controllers
+{
+    public class SettingsController : Controller
+    {
+        public IActionResult Index()
+        {
+            return View();
+        }
+
+        [HttpPost]
+        public IActionResult ChangePassword(UserModel model)
+        {
+            model.Username = "test";
+
+            if (model.CurrentPassword.Equals(model.Password))
+            {
+                if (model.NewPassword.Equals(model.Password))
+                {
+                    //New password cannot be the same as old one.
+                    return View("../Settings/PasswordChangeError", model);
+                }
+                else if (model.NewPassword.Equals(model.NewPassword2))
+                {
+                    model.ChangePassword(model.NewPassword);
+                    return View("../Settings/PasswordChanged", model);
+                }
+                else
```

Comment:
Password match validation can be done in view. Check Registration Index view. Line 33.

Reply:
Great idea, thanks.